src/darwin/agent/launchd.js

```diff
@@ -121,6 +127,26 @@ Interceptor.attach(Module.getExportByName('/usr/lib/system/libsystem_kernel.dyli
   }
 });
 
+function isPrewarmLaunch(envp) {
+  if (envp.isNull())
+    return null;
+
+  let cur = envp;
+  while (true) {
+    const elementPtr = cur.readPointer();
+    if (elementPtr.isNull())
+      break;
+
+    const element = elementPtr.readUtf8String();
+    if (element.startsWith('ActivePrewarm'))
+      return true;
+
+    cur = cur.add(pointerSize);
+  }
+
+  return false;
+}
```

Comment:
yeah i was in doubt because in this way we would give up terminating the loop early but, as you said, most of the time we iterate it all at least once, so yeah parsing it properly once would be more readable and performance won't be worse and can be better in some cases.

on it